docs/src/main/asciidoc/reaugmentation.adoc

```diff
@@ -0,0 +1,66 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Reaugment a Quarkus Application
+
+include::./attributes.adoc[]
+
+== What is augmentation?
+
+During augmentation the application is initialized - or let's say pre-initialized.
+Things that used to happen when an EAR file was deployed on a Jakarta-EE server such as parsing static configuration, creating proxy instances, etc. now happen at augmentation time.
+If you add CDI beans after augmentation they won't work (because of the missing proxy classes) and if you change build-time properties (e.g. `quarkus.datasource.db-kind`) after augmentation these changes will be ignored.
+Build-time properties are marked with a lock in the list of link:all-config[All Configuration Options].
+It doesn't matter if you use profiles or any other way to override the properties.
+The build-time properties that were active during augmentation are baked into the build.
+
+> Reaugmentation allows you to recreate the augmentation output for a different build configuration
+
+== When is reaugmentation useful?
+
+Reaugmentation is useful if you ship your software to multiple users which want to be able to change some build-time properties.
+For instance changing the database driver or switching features on or off (e.g. jaeger).
+If there are only two or three possible settings, you can easily provide multiple builds. But if the possible combinations get more you probably prefer shipping a mutable jar (the number of combinations of n switches is 2^n).
```

Comment:
```suggestion
If there are only two or three build-time properties that depend on the user environment, you may consider providing alternative builds of the application. However, in case there are more such properties you may prefer shipping a mutable jar instead and let your users re-augment the application for their environment.
```